Fix sign comparison

r-barnes opened the pull request:
Differential Revision: D29885396

albanD commented on aten/src/ATen/native/cuda/CuFFTPlanCache.h:
```diff
@@ -507,7 +508,7 @@ class CuFFTParamsLRUCache {
     TORCH_CHECK(new_size >= 0,
              "cuFFT plan cache size must be non-negative, but got ", new_size);
-    TORCH_CHECK(new_size <= CUFFT_MAX_PLAN_NUM,
+    TORCH_CHECK(new_size <= static_cast<int64_t>(CUFFT_MAX_PLAN_NUM),
```
Should we just make CUFFT_MAX_PLAN_NUM a int64_t from the beginning? There is a static assert that it is always positive already.

r-barnes replied:
I'd support that. Signed types seem to be generally easier to work with; however, I suspect we'll still have inappropriate comparisons. Let me know what works best for you.

albanD replied:
Given that these values can never really overflow a signed int. I would lean towards whatever reduces the number of casts the most :D
But if you think this is cleaner, I'm happy with that as well.

albanD commented on torch/csrc/autograd/custom_function.h:
```diff
@@ -315,12 +315,10 @@ variable_list CppNode<T>::apply(variable_list&& inputs) {
-  // NOLINTNEXTLINE(cppcoreguidelines-init-variables)
-  int num_forward_inputs = is_variable_input_.size();
-  auto num_outputs = outputs.size();
+  const auto num_forward_inputs = static_cast<int64_t>(is_variable_input_.size());
```
Why can't these two stay size_t?

r-barnes replied:
`num_outputs` comes from `auto outputs = T::backward(&ctx_, backward_inputs);` where `T` is a template. I suspect the type of `outputs.size()` varies for different `T`.

albanD replied:
Ho interesting. In my mind we were enforcing that backward always returns a variable_list. But that is not the case indeed.

r-barnes committed:
cuFFT plan fix signs (#62194)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/62194

Differential Revision: D29885396

fbshipit-source-id: e256eacad90ad5ee8d72cf8e6966b2c3d7ae95de
